### ISSUE TITLE
[$250] IOU - Distance tab flickering to manual & back to distance every time when add destination

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#46346]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.13-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** gocemate+a753@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Login with gmail account
2. Go to FAB> Submit Expense> Distance
3. Add a start destination> Your location
4. Check tabs (manual, scan, distance) behavior
5. Add stop destination
6. Check tabs (manual, scan, distance) behavior

## Expected Result:
Distance tab should remain since we work on that tab

## Actual Result:
Distance tab switch to manual and back to distance every time when add destination

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/f07d9d5c-a93a-4c22-9df8-f10c49633c19

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b7f69f2e00762143</li>
        <li>Upwork Job ID: 1817973479806682768</li>
        <li>Last Price Increase: 2024-08-08</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @cdOut</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vip

**Comment by lanitochka17** [comment]
@Christinadobrzyn  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by Christinadobrzyn** [comment]
I can reproduce this - only seems to be affecting Android web app. I think this can be external. I _think_ this might be part of Collect since it's a distance expense.

**Comment by Christinadobrzyn** [comment]
waiting on proposals

**Comment by Christinadobrzyn** [comment]
I'll see if someone at SWM would want this.

**Comment by Christinadobrzyn** [comment]
I'll see if someone at SWM would want this. https://expensify.slack.com/archives/C04878MDF34/p1722447232998509

**Comment by Christinadobrzyn** [comment]
I'll see if someone at SWM would want this. https://expensify.slack.com/archives/C04878MDF34/p1722447232998509

**Comment by cdOut** [comment]
Hi, I’m Tymoteusz from Software Mansion, the expert agency, and I’m going to be working on this task.

**Comment by Christinadobrzyn** [comment]
Awesome! Thank you @cdOut! I've assigned you to the GH.

**Comment by VincentCorleone** [comment]
> Hi, I’m Tymoteusz from Software Mansion, the expert agency, and I’m going to be working on this task.

@cdOut So can external contributors submit a proposal and get paid from upwork?

**Comment by cdOut** [comment]
@VincentCorleone I'm taking over this issue so no, there is no need for proposals.

**Comment by cdOut** [comment]
I'm still figuring it out, PR should be up today or tomorrow.

**Comment by cdOut** [comment]
Had a slight delay by another issue getting unblocked by backend merges, will continute today.

**Comment by Christinadobrzyn** [comment]
Hi @cdOut just checking in on this to see you have an update before the weekend. Thank you!

**Comment by cdOut** [comment]
I'm working on the PR for this issue, draft will be up early next week.

**Comment by cdOut** [comment]
cc: @Christinadobrzyn

We've investigated this issue further and discovered that it not only affects other platforms (page rendering on web and desktop is smoother, which is why it's difficult to catch the flicker as it happens very quickly) but also impacts the `Start chat` flow, since it uses the same components.

We have identified a working fix: the problem appears to reside in the `react-native-tab-view` library. This library makes the selected tab index dependent on the view layout, and the view returns a `width` and `height` of 0 for a split second when navigating to a page containing the top bar, before obtaining the correct dimensions from an event listener. This results in it always pointing to the first element no matter which one is actually selected for that split second.

I will search for a way to fix this within our codebase rather than relying on an external library. If that proves unfeasible, I will create a PR to address this issue in the library and proceed to implement a temporary patch in our codebase.

**Comment by Christinadobrzyn** [comment]
Update for Melvin - @cdOut is searching for a fix based on this [update](https://github.com/Expensify/App/issues/46346#issuecomment-2284075466). 

Can you let us know if there's more of an update @cdOut?

**Comment by Christinadobrzyn** [comment]
I'm just checking in on this @cdOut. If you can provide an update, that would be great! Thank you!

**Comment by cdOut** [comment]
@Christinadobrzyn I'm going to be raising a draft PR with a fix patch for it in a moment.

**Comment by Christinadobrzyn** [comment]
Hi, @cdOut, I'm just checking on this to see if you have any specific updates - otherwise, it looks like the PR is in the works!

**Comment by cdOut** [comment]
No update needed, PR with the patch is being tested by Manan.

**Comment by thienlnam** [comment]
Just merged the PR, but let's keep this issue open until we get the patch removed with an upstream PR

**Comment by Christinadobrzyn** [comment]
PR deployed to production yesterday - https://github.com/Expensify/App/pull/47885 so adding the payment date to the subject. 

I'm going to be ooo 9/4 - 9/10 so I'll plan on paying this when I'm back - if that's okay. 

Do we need a regression?
@mananjadhav are you paid through NewDot?

Payouts due:

- [ ] Contributor+: $250 @mananjadhav

**Comment by cdOut** [comment]
Please check the conversation in the [PR itself here](https://github.com/Expensify/App/pull/47885#issuecomment-2322027791), this patch sadly broke another functionality that we didn't catch during testing so we had to revert it.

I'm going to be working on a fix that doesn't break the tab drag changing animation so please keep this issue open until a new PR gets merged.

**Comment by cdOut** [comment]
Hi, wanted to update this issue. I currently have to focus on other more important issues and I will pick this issue back up early next week.

**Comment by mananjadhav** [comment]
Thanks for the update @cdOut. Is there someone else who can pick this up? or you'll pick this once you're done with the other PRs?

**Comment by Christinadobrzyn** [comment]
I'm back - looks like we're working on another PR? @cdOut can you provide the new PR to follow?

**Comment by cdOut** [comment]
It's not raised yet, I'll let you know when I have it up.

**Comment by cdOut** [comment]
Hi, quick update - I've picked back the issue and I'm searching for a solution that won't affect the side swiping on tabs, should raise a PR for it in the coming days.

**Comment by cdOut** [comment]
Short update, this is a pretty convoluted issue as fixing the flicker and allowing for the swipe causes another animation in tabs to break so I'm still researching a fix that won't disturb any other parts of the flow.

**Comment by Christinadobrzyn** [comment]
Just a heads up that I'm going to be ooo until 9/30. I'm not going to assign another BZ teammate since we're working on the fix side of this. If you need anyone please reach out in Slack for a volunteer.

**Comment by Christinadobrzyn** [comment]
payment summary here - https://github.com/Expensify/App/issues/46346#issuecomment-2327631847

**Comment by mananjadhav** [comment]
@Christinadobrzyn this was reverted and the follow up PR is still pending. No payment yet on this one.

**Comment by mananjadhav** [comment]
Thanks for the update @cdOut

**Comment by cdOut** [comment]
I've found some discrepancies with animations while testing on native devices, will work on correcting it throughout today.

**Comment by mananjadhav** [comment]
@cdOut Quick bump on this one.

**Comment by cdOut** [comment]
Yes, sorry for not updating this thread. I got dragged back into debugging last issues with a more critical Guided Setup task, then I kept working on a possible update to the library both on their repo and on ours, an implementation with a fix that doesn't touch the animations when dragging will be up and out of draft in a PR in the coming days.

**Comment by cdOut** [comment]
Hi, I should have the draft up for review by the EOW.

**Comment by cdOut** [comment]
I've re-checked it during looking for a fix on the latest main branch, and it looks like there were changes made in the library itself which fixed the issue. The recording below is from the latest main on android mWeb, looks like the issue has been resolved.

@Christinadobrzyn @mananjadhav If you could also re-test it on your end just to make sure the issue has been fixed. I've checked it both for android native and mWeb, the flicker doesn't appear anymore when opening the tab / navigating back to it from a subsection.

https://github.com/user-attachments/assets/e8bb3eb8-a3b3-48db-b340-e0d568700c27

**Comment by blazejkustra** [comment]
@cdOut Can you link a PR in the upstream repo that fixed this issue? (or version of the library which fixed it)

**Comment by cdOut** [comment]
I've sifted through the recent PRs and it looks like there was a similar problem reported discussing the `Submit expense` tab which also fixed along this issue as well. The PR for it is [here](https://github.com/Expensify/App/pull/51020). Looks like Kuba decided to refactor the animations for the tab selector from the ground up.

**Comment by mananjadhav** [comment]
I can't reproduce it either. @Christinadobrzyn Does this need any further action?

If not, I would request you to consider partial payout for the work here. I reviewed a PR that ended up being reverted. Hence, requesting small partial payout.

**Comment by Christinadobrzyn** [comment]
I think we can close this if we can't reproduce it but @thienlnam, can you double-check that we can close this? I'm not sure if it should have QA test again?

**Comment by thienlnam** [comment]
I'll drop it in the QA room to have them re-test so we can confirm

**Comment by kavimuru** [comment]
Issue not reproducible anymore

https://github.com/user-attachments/assets/53e85896-eff5-455d-ad81-c188a4a1c1e3

**Comment by mananjadhav** [comment]
@Christinadobrzyn should we close it then? and also my request [here](https://github.com/Expensify/App/issues/46346#issuecomment-2452383925) ?

**Comment by Christinadobrzyn** [comment]
Yes, let's close. Since a PR was reviewed but also reverted, we can make a partial payment of 50%. 

Payouts due:

- [ ] Reviewer: $125 @mananjadhav 

Are you paid through NewDot?

**Comment by mananjadhav** [comment]
Yes paid via NewDot.

**Comment by Christinadobrzyn** [comment]
Sounds good! Thanks for all the work @cdOut and @mananjadhav! 

Feel free to request the ND payment @mananjadhav - we'll close this.

**Comment by garrettmknight** [comment]
$125 approved for @mananjadhav
